### DISCUSSION TITLE
05.12.2021

### DISCUSSION THREAD

**Discussion by farag2** [farag2/Sophia-Script-for-Windows#306] [General]
# Before running the script read [Before running](https://github.com/farag2/Sophia-Script-for-Windows#fire-before-running-fire) & [How to use](https://github.com/farag2/Sophia-Script-for-Windows#how-to-use)

[![YT](https://img.youtube.com/vi/f529ucAipI8/0.jpg)](https://youtu.be/f529ucAipI8) [![YT](https://img.youtube.com/vi/MiQ85tVXQQA/0.jpg)](https://youtu.be/MiQ85tVXQQA)

***

[![ko-fi](https://www.ko-fi.com/img/githubbutton_sm.svg)](https://ko-fi.com/Q5Q51QUJC)

<a href="https://yoomoney.ru/to/4100116615568835"><img src="https://yoomoney.ru/i/shop/iomoney_logo_color_example.png" width=220px height=46px></a>

***

[![actions](https://img.shields.io/badge/Sophia%20News-Telegram-blue?style=flat&logo=Telegram)](https://t.me/SophiaNews) [![actions](https://img.shields.io/badge/Sophia%20Chat-Telegram-blue?style=flat&logo=Telegram)](https://t.me/Sophia_Chat) [![actions](https://img.shields.io/github/workflow/status/farag2/Sophia-Script-for-Windows/Build?label=GitHub%20Actions&logo=GitHub)](https://github.com/farag2/Sophia-Script-for-Windows/actions)

## Windows 11 21H2 | Windows 10 2004/20H2/21H1/21H2 | Enterprise LTSC 2021 | Enterprise LTSC 2019

* Now you can download the always latest Sophia Script archive by invoking (`not as administrator too`) in PowerShell console

```powershell
irm script.sophi.app | iex
```

* The command will download and expand the latest Sophia Script archive (`without running`) according which Windows and PowerShell versions it is run on. If you run it on Windows 11 via PowerShell 5.1, it will download Sophia Script for `Windows 11 PowerShell 5.1`.

Diff from v6.0.7
[6.0.7...6.0.8](https://github.com/farag2/Sophia-Script-for-Windows/compare/6.0.7...6.0.8)

* Fixed small bug in the `OneDrive` function that prevents a folder to be removed;
* Fixed small bug in the `OneDrive` function that prevents OneDrive to be downloaded (Microsoft changed the cloud XML silently);
* The `WinPrtScrFolder` function was improved;
  * Now it doesn't matter how your preset was named: the fucntion will parse the preset that was loaded into session to find whether the `OneDrive -Uninstall` function was commented out or not;
* Minor changes.
* [Check out](https://t.me/SophiaNews/396) the 4th [SophiApp](https://github.com/Sophia-Community/SophiApp) 0.0.0.70 public alpha build :rocket:

![SophiApp](https://hsto.org/r/w780/getpro/habr/upload_files/be9/060/0b6/be90600b648639aa85d755fe10677cb2.jpg)

## Sophia Script Wrapper 2.5.3

* More validation for JSON added;
* Backup of `Sophia.ps1` to `Sophia-original.ps1` if using import ps1 file called `Sophia.ps1`;
  * Otherwise if importing `x.ps1`, the file will be overwritten if the `Run PowerShell` button is used.

![Wrapper](https://i.imgur.com/yS0eESG.png)

|              File                                     |                          SHA-256                               |
|:-----------------------------------------------------:|:--------------------------------------------------------------:|
| Sophia.Script.for.Windows.10.LTSC.2019.v5.2.17.zip    |11B8D748FEDC1DEFF15BED8CE1A0F6CF90D2347A8F83FA172C5B75D9B8A2C676|
| Sophia.Script.for.Windows.10.LTSC.2021.v5.12.7.zip    |7EFDF531E4826CD43228EE33F2FC0583AB94C99E46EE7BC0C94BD38B28160048|
| Sophia.Script.for.Windows.10.v5.12.7.PowerShell.7.zip |A26FFA058C167D5480F82F97534BEDF26FCF73256EAF93D4A6935D17991E82A0|
| Sophia.Script.for.Windows.10.v5.12.7.zip              |6C864477FFB65AD357F9846B356E76CF949564EE4F337C34470765CDFCBA5501|
| Sophia.Script.for.Windows.11.v6.0.8.PowerShell.7.zip  |A6336109300ECA5037E9DE294F922178A669E21DA0A2B8063EB1C928AB1F7588|
| Sophia.Script.for.Windows.11.v6.0.8.zip               |3E7A276E2E806DD3DFEEE338F5286EF8AF071CFBA58A7B5020E000C61A966FE5|
| Sophia.Script.Wrapper.v2.5.3.zip                      |494E6DD8B39783488E7D1DB484CD14ED3AAEB2BD2ED3FFF16E91B1FE5E1467F8|

<hr /><em>This discussion was created from the release <a href='https://github.com/farag2/Sophia-Script-for-Windows/releases/tag/6.0.8'>05.12.2021</a>.</em>

**Comment by f00tf00t** [comment]
Hi there,

Within the zip file for Sophia.Script.Wrapper.v2.5.3.zip the folder is named Sophia.Script.Wrapper.v2.5.2

**Reply by farag2** [reply]
Everything is possible). I'll fix all bugs left if the latest build later this year. I gope, it will be the last for this year.

**Reply by farag2** [reply]
Re-uploaded. 😸
